src/main/java/com/iota/iri/LedgerValidator.java

```diff
@@ -171,17 +164,13 @@ private void updateSnapshotMilestone(MilestoneViewModel milestone) throws Except
      * @param tip
      * @throws Exception
      */
-    private void updateConsistentHashes(final Snapshot snapshot, Hash tip, int index) throws Exception {
+    private void updateConsistentHashes(final Set<Hash> visitedHashes, Hash tip, int index) throws Exception {
```

Comment:
this + the fact that `visitedHashes` is not updated in `updateDiff` anymore probably means you are missing a call to `updateConsistentHashes`

Reply:
I was missing a call, thanks - but since this would be redundant, I refactored to just use a copy of the set fed into getLatestDiff, and update it if consistent.